src/main/java/frc/robot/commands/AlignToPose.java

```diff
@@ -0,0 +1,109 @@
+package frc.robot.commands;
+
+import static edu.wpi.first.units.Units.MetersPerSecond;
+import static edu.wpi.first.units.Units.Radians;
+
+import com.ctre.phoenix6.swerve.SwerveModule.DriveRequestType;
+import com.ctre.phoenix6.swerve.SwerveRequest;
+import dev.doglog.DogLog;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.generated.TunerConstants;
+import frc.robot.subsystems.CommandSwerveDrivetrain;
+import java.util.function.Supplier;
+
+public class AlignToPose extends Command {
+  Pose2d currPose;
+  double angularVelocity;
+  DriveCommand driveCommand;
+  Supplier<Pose2d> targetPose;
+  private PIDController PIDX;
+  private PIDController PIDY;
+  private PIDController PIDRotation;
+  private CommandSwerveDrivetrain drivetrain;
+  private double MaxSpeed =
+      TunerConstants.kSpeedAt12Volts.in(MetersPerSecond); // kSpeedAt12VoltsMps desired top speed
+  private double MaxAngularRate =
+      3.5 * Math.PI; // 3/4 of a rotation per second max angular velocity
+  private final SwerveRequest.FieldCentric drive =
+      new SwerveRequest.FieldCentric()
+          .withDeadband(MaxSpeed * 0.1)
```

Comment:
we may need to remove the Deadband here.   It is useful for when we use the joystick, but maybe not for being driven by code.  It will squash the small values to zero, which may not be right.